src/Client.cpp

```diff
@@ -874,15 +874,24 @@ void CClient::HandleCap(const CString& sLine)
 	//TODO support ~ and = modifiers
 	CString sSubCmd = sLine.Token(1);
 
+	std::map<CString, std::function<void(bool bVal)>> mCoreCaps = {
```

Comment:
Yep, that's good. I was just thinking that the map value could've been a simple bool ptr/ref, but perhaps there will be caps with more complex handler functions in the future.